### PR TITLE
interpreters/bas: Fix the nxstyle warning

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

except "Mixed case identifier found"

## Impact

Minor

## Testing

CI

### CODE REVIEW

**Comment by acassis** [comment]
Does sim:bastest still working after this PR?

**Comment by xiaoxiang781216** [comment]
> Does sim:bastest still working after this PR?

Yes.
```
NuttShell (NSH) NuttX-12.0.0
nsh> bas
bas 2.4
Copyright 1999-2014 Michael Haardt.
This is free software with ABSOLUTELY NO WARRANTY.
> 
> 
> help
Error: Undeclared function or variable at: help
> ?

> 4+5
> print 4+5
 9 
```

File: interpreters/bas/bas_value.c [hunk 1]
```diff
@@ -2026,9 +2042,21 @@ struct String *Value_toWrite(struct Value *this, struct String *s)
...
+  {
+    V_REAL
+  };
+
+  static struct Value string =
+  {
+    V_STRING
+  };
+
   static char n[] = "";
```
Comment (pkarashchenko) [inline interpreters/bas/bas_value.c:2060]: Can we remove `n` and use `""` directly?

Comment (xiaoxiang781216) [reply interpreters/bas/bas_value.c]: Done.

File: interpreters/bas/bas_value.c [hunk 2]
```diff
@@ -2026,18 +2042,29 @@ struct String *Value_toWrite(struct Value *this, struct String *s)
...
+  {
+    V_REAL
+  };
+
+  static struct Value string =
+  {
+    V_STRING
+  };
+
   static int init = 0;
```
Comment (pkarashchenko) [inline interpreters/bas/bas_value.c:2060]: Also I do not see where `init` is set to 1. Is that needed at all? Can we go with static initialization for all the structures?

Comment (xiaoxiang781216) [reply interpreters/bas/bas_value.c:2060]: > Also I do not see where `init` is set to 1. Is that needed at all? Can we go with static initialization for all the structures?

No, struct Value contain union, it's hard to initialize union with c89 syntax. So, I add code to set init to 1

**Review by pkarashchenko — APPROVED** [review]

File: interpreters/bas/bas_value.c (hunk 2 above)
Comment (xiaoxiang781216) [reply interpreters/bas/bas_value.c:2060]: but the init value point to a string which is an null string, not zero.

Comment (pkarashchenko) [reply interpreters/bas/bas_value.c:2060]: Yeap. Missed that. So there is no other way.

**Pull request merged**